### PR TITLE
[RTM] Auto configure the commands

### PR DESCRIPTION
**Pull request by leofeyer**
This PR implements #799 for the commands. (There are a few unrelated changes which are necessary to make the bundle work with Symfony 3.3.)

### CODE REVIEW

File: src/Resources/config/commands.yml [hunk 1]
```diff
@@ -1,32 +1,26 @@
 services:
+    _defaults:
+        autoconfigure: true
```
Comment (Toflar) [inline src/Resources/config/commands.yml]: Can we add a `# Automatically tags commands as console.command` here and whenever we add something new that uses the `autoconfigure` extend this comment? Just for future reference and transparency :)

**Comment by leofeyer** [comment]
> Can't we also use the autoconfiguration for `kernel.cache_clearer`, `kernel.cache_warmer` and others?

Yes, of course. But this PR only includes the commands. 😉

**Comment by Toflar** [comment]
Ah shit, sorry I thought this was a complete SF 3.3 change. Reading titles would help. My bad!

File: src/Resources/config/commands.yml (hunk 1 above)
Comment (leofeyer) [inline src/Resources/config/commands.yml]: I guess it is pretty obvious what the option does in the `commands.yml` file, isn't it?

Comment (Toflar) [inline src/Resources/config/commands.yml]: Yes, I just approved the PR. This comment was related to my initial thought that it should be for everything.

**Comment by leofeyer** [comment]
For some reason it did not work with the trait. I tried this first. 😄

**Pull request merged**